www_src/pages/project/Positionable.jsx

```diff
@@ -0,0 +1,135 @@
+"use strict";
+
+var React = require("react");
+var classes = require("classnames");
+
+var Positionable = React.createClass({
+
+  getInitialState: function() {
+    return {
+      x: this.props.x || 0,
+      y: this.props.y || 0,
+      scale: this.props.scale || 1,
+      angle: this.props.angle || 0,
+      zIndex: (typeof this.props.zIndex !== "undefined") ? this.props.zIndex : 1,
+      interactive: (typeof this.props.interactive !== "undefined") ? this.props.interactive : true,
+      touchactive: false
+    };
+  },
+
+  componenWillMount: function() {
+    if(!this.state.interactive) return;
+
+    this.markx = this.state.x;
+    this.marky = this.state.y;
+    this.markangle = this.state.angle;
+    this.markscale = this.state.scale;
+  },
+
+  componentDidMount: function() {
+    if(!this.state.interactive) return;
+    var touchHandler = this.touchhandler = require("./touchhandler")(this);
+
+    var dnode = this.getDOMNode();
+    dnode.addEventListener("touchstart", touchHandler.startmark);
+    dnode.addEventListener("mousedown", touchHandler.startmark);
+    dnode.addEventListener("touchmove", touchHandler.panmove);
+    dnode.addEventListener("mousemove", touchHandler.panmove);
+    dnode.addEventListener("touchend", touchHandler.endmark);
+    dnode.addEventListener("mouseup", touchHandler.endmark);
+
+    var onode = this.refs.overlay.getDOMNode();
+    onode.addEventListener("touchstart", touchHandler.secondFinger);
+    onode.addEventListener("touchmove", touchHandler.handleRS);
+    onode.addEventListener("touchend", touchHandler.endSecondFinger);
+  },
+
+  componentDidUpdate: function(prevProps, prevState) {
+    if(!prevState.touchactive && this.state.touchactive) {
+      this.props.onUpdate(this.state);
+    }
+  },
+
+  render: function() {
+    var x = this.state.x,
+        y = this.state.y,
+        angle = this.state.angle,
+        scale = this.state.scale,
+        zIndex = this.state.zIndex;
+
+    var style = {
+      transform: [
+        "translate("+x+"px, "+y+"px)",
+        "rotate("+angle+"deg)",
+        "scale("+scale+")"
+      ].join(" "),
+      transformOrigin: "center",
+      zIndex: zIndex
+    };
+
+    var className = classes({
+      positionable: true,
+      touchactive: this.state.touchactive,
+      current: this.props.current
+    });
+
+    return (
+      <div>
+        <div ref="overlay" className="touchOverlay" hidden={!this.state.touchactive} />
```

Comment:
remember that JSX is **not** HTML. JSX allows self closing syntax on all XML notation for components.